File: test/Button/test03.c
#include <libming.h>

int main()
{
	SWFMovie m = newSWFMovie();

	SWFShape s = newSWFShape();
	SWFShape_setLine(s, 10, 255, 0, 0, 255);
	SWFFillStyle fill = newSWFSolidFillStyle(255, 255, 0, 255);
	SWFShape_setRightFillStyle(s, fill);
	SWFShape_movePenTo(s, -50, -20);
	SWFShape_drawLine(s, 100, 0);
	SWFShape_drawLine(s, 0, 40);
	SWFShape_drawLine(s, -100, 0);
	SWFShape_drawLineTo(s, -50, -20);

	// simple animation by relative scaling
	SWFMovieClip mc1 = newSWFMovieClip();
	SWFDisplayItem si1 = SWFMovieClip_add(mc1, (SWFBlock) s);
	SWFMovieClip_nextFrame(mc1);
	SWFDisplayItem_scale(si1, 0.8, 0.8);
	SWFMovieClip_nextFrame(mc1);
	SWFDisplayItem_scale(si1, 0.8, 0.8);
	SWFMovieClip_nextFrame(mc1);
	SWFDisplayItem_scale(si1, 0.8, 0.8);
	SWFMovieClip_nextFrame(mc1);
	SWFDisplayItem_scale(si1, 0.8, 0.8);
	SWFMovieClip_nextFrame(mc1);
	SWFDisplayItem_scale(si1, 0.8, 0.8);
	SWFMovieClip_nextFrame(mc1);

	// another animation by absolute scaling
	SWFMovieClip mc2 = newSWFMovieClip();
	SWFDisplayItem si2 = SWFMovieClip_add(mc2, (SWFBlock) s);
	SWFDisplayItem_scaleTo(si2, 0.2, 0.2);
	SWFMovieClip_nextFrame(mc2);
	SWFDisplayItem_scaleTo(si2, 0.4, 0.4);
	SWFMovieClip_nextFrame(mc2);
	SWFDisplayItem_scaleTo(si2, 0.6, 0.6);
	SWFMovieClip_nextFrame(mc2);
	SWFDisplayItem_scaleTo(si2, 1, 1);
	SWFMovieClip_nextFrame(mc2);

	SWFButton b = newSWFButton();

	SWFButton_addCharacter(b, (SWFCharacter)s, SWFBUTTON_HIT|SWFBUTTON_UP);
	SWFButton_addCharacter(b, (SWFCharacter)mc1, SWFBUTTON_OVER);
	SWFButton_addCharacter(b, (SWFCharacter)mc2, SWFBUTTON_DOWN);

	SWFDisplayItem d;
	d = SWFMovie_add(m, b);
	SWFDisplayItem_moveTo(d, 150, 100);

	SWFMovie_nextFrame(m);
	SWFMovie_save(m, "test03.swf");
	
	return 0;
}

File: test/Button/test03.pl
#!/usr/bin/perl -w

use strict;
use SWF (':ALL');
use SWF::Constants (':Button');
my $m = new SWF::Movie;

my $s = new SWF::Shape;
$s->setLine(10, 255, 0, 0, 255);
$s->setRightFill(255, 255, 0, 255);
$s->movePenTo(-50,-20);
$s->drawLine(100, 0);
$s->drawLine(0, 40);
$s->drawLine(-100, 0);
$s->drawLineTo(-50, -20);

# animation by relative scaling
my $mc1= new SWF::MovieClip;
my $si1=$mc1->add($s);
$mc1->nextFrame;
$si1->scale(0.8);
$mc1->nextFrame;
$si1->scale(0.8);
$mc1->nextFrame;
$si1->scale(0.8);
$mc1->nextFrame;
$si1->scale(0.8);
$mc1->nextFrame;
$si1->scale(0.8);
$mc1->nextFrame;

# another animation by absolute scaling
my $mc2= new SWF::MovieClip;
my $si2=$mc2->add($s);
$si2->scaleTo(0.2);
$mc2->nextFrame;
$si2->scaleTo(0.4);
$mc2->nextFrame;
$si2->scaleTo(0.6);
$mc2->nextFrame;
$si2->scaleTo(1);
$mc2->nextFrame;

my $b = new SWF::Button;
my $br1 = $b->addCharacter($s, 
	SWF::Constants::SWFBUTTON_HIT|SWF::Constants::SWFBUTTON_UP);
my $br2 = $b->addCharacter($mc1, SWF::Constants::SWFBUTTON_OVER);
my $br3 = $b->addCharacter($mc2, SWF::Constants::SWFBUTTON_DOWN);

my $d = $m->add($b);
$d->moveTo(150, 100);

$m->nextFrame;
$m->save("test03.swf");
